python/lsst/ts/externalscripts/base_take_twilight_flats.py

```diff
@@ -0,0 +1,457 @@
+        """
+        Returns the RADEC of the target area of the sky that's an azimuth
+        `distance_from_sun` away from the Sun, given `elevation`,
+        and at a given `time`.
+
+        Parameters
+        ----------
+        distance_from_sun : float, (-180, 180)
+            The distance from the Sun in degrees.
+            Positive angles go towards the North.
+        elevation : float
+            Target elevation for Sky Flats.
+        """
+
+        az_sun, el_sun = self.tcs.get_sun_azel()
+
+        target_az = (az_sun + self.config.distance_from_sun) % 360
+
+        target_radec = self.tcs.radec_from_azel(target_az, self.config.target_el)
+
+        return target_radec
+
+    def get_empty_field(self, target, radius=5):
+        """
+        Query the "Deep blank field catalogue : J/MNRAS/427/679" in Vizier.
+
+        Parameters
+        ----------
+        target : astropy.coordinates.SkyCoord
+            Sky coordinates near the field
+        radius : float
+            Search radius in degrees.
+
+        Reference
+        ---------
+        http://cdsarc.u-strasbg.fr/viz-bin/Cat?J/MNRAS/427/679
+        """
+        _table = Vizier.query_region(
+            catalog="J/MNRAS/427/679/blank_fld",
+            coordinates=target,
+            radius=radius * u.deg,
+        )
+
+        if len(_table) == 0:
+            self.log.info(
+                f"Could not find a field near {target} " f"within {radius} deg radius"
+            )
+            return None
+
+        _table = _table["J/MNRAS/427/679/blank_fld"]
+
+        coords = coordinates.SkyCoord(
+            ra=_table["RAJ2000"],
+            dec=_table["DEJ2000"],
+            unit=(u.hourangle, u.deg),
+            frame=coordinates.ICRS,
+        )
+
+        arg = target.separation(coords).argmin()
+
+        return coords[arg]
+
+    def assert_sun_location(self):
+        """Confirm sun's elevation is safe for taking twilight flats."""
+        sun_coordinates = self.tcs.get_sun_azel()
+        where_sun = "setting" if (sun_coordinates[0] > 180) else "rising"
+        self.log.debug(
+            f" The azimuth of the {where_sun} Sun is {sun_coordinates[0]:.2f} deg \n"
+            f" The elevation of the Sun is {sun_coordinates[1]:.2f} deg"
+        )
+
+        if (sun_coordinates[1] < self.config.min_sun_elevation) or (
+            sun_coordinates[1] > self.config.max_sun_elevation
+        ):
+            raise AssertionError(
+                f"Sun elevation {sun_coordinates} is outside appropriate elvation limits. Aborting."
+            )
+
+    async def take_twilight_flats(self):
+
+        group_id = self.group_id if self.obs_id is None else self.obs_id
+
+        self.assert_sun_location()
+
+        target = self.get_target_radec()
+
+        # get an empty field
+        search_area_degrees = 10
+
+        empty_field_coords = self.get_empty_field(target, radius=search_area_degrees)
+        if empty_field_coords is not None:
```

Comment:
you could simplify all this by handling this situation inside `get_empty_field`. 

Instead of returning `None`, if there's no field, or the field coordinate it there is one, return ra/dec directly.

So here you would do:

```
ra, dec = await self.get_empty_field(target, radius=search_area_degrees)
```

I would, however, rename this method to something like `get_twilight_flat_sky_coordinates`.